### PR TITLE
log: replace duplicated version with the configurable name

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
KRKNWK-19484
remove version, as it is already printed in logs, and replace it with application name that can be configured with KConfig.

## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: handling-APP_NAME
```

## Description

JIRA ticket: 

## Self review

- [x] There is no commented code.
- [x] There are no TODO/FIXME comments without associated issue ticket.
- [x] Commits are properly organized.
- [x] Change has been tested.
- [x] Tests were updated (if applicable).

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -100,6 +100,13 @@ config SID_HAL_PROTOCOL_MEMORY_SZ
 	int
 	default 1024
 
+config SIDEWALK_APPLICATION_NAME
+	string "Name of the application printed on start up"
+	default ""
```
Comment (ktaborowski) [inline Kconfig:105]: consider setting the defautl name linke "custom_app" or sth.  The "APP NAME" string will pirint the name, even when it's empty.

File: utils/include/sidewalk_version.h [hunk 2]
```diff
@@ -14,7 +14,8 @@
 	LOG_PRINTK("----------------------------------------------------------------\n");          \
 	LOG_PRINTK("Sidewalk SDK        = %d.%d.%d.%d\n", SID_SDK_MAJOR_VERSION,                   \
 		   SID_SDK_MINOR_VERSION, SID_SDK_PATCH_VERSION, SID_SDK_BUILD_VERSION);           \
-	LOG_PRINTK("APP_BUILD_VERSION   = %s\n", STRINGIFY(APP_BUILD_VERSION));                    \
+	LOG_PRINTK("APP_BUILD_VERSION   = %s\n", STRINGIFY(APP_BUILD_VERSION));			   \
+	LOG_PRINTK("APP_NAME            = %s\n", CONFIG_SIDEWALK_APPLICATION_NAME);		   \
```
Comment (ktaborowski) [inline utils/include/sidewalk_version.h:18]: Is'nt it redundant, when `CONFIG_NCS_BOOT_BANNER` is enabled? You can use `CONFIG_NCS_APPLICATION_BOOT_BANNER_STRING` (and the build version is a git commit hash)

Comment (RobertGalatNordic) [reply utils/include/sidewalk_version.h:18]: ok, that is a mistake, I intended to replace the APP_BUILD_VERSION with APP_NAME but I see I just added APP_NAME.

File: Kconfig (hunk 1 above)
Comment (RobertGalatNordic) [reply Kconfig:105]: ok

File: utils/include/sidewalk_version.h (hunk 2 above)
Comment (totyz) [reply utils/include/sidewalk_version.h:18]: I intentional leave APP_BUILD_VERSION as it's required for tests

File: Kconfig (hunk 1 above)
Comment (totyz) [reply Kconfig]: set default to "sidewalk"

**Review by totyz — APPROVED** [review]
All looks OK.
Test are passing

**Pull request merged**